### PR TITLE
Feature/page with users event types

### PR DESCRIPTION
**Pull request by dmitriy-uvin**

### CODE REVIEW

File: api/app/Actions/EventType/GetEventTypeCollectionByNicknameAction.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        $this->eventRepository = $eventRepository;
+        $this->userRepository = $userRepository;
+    }
+
+    public function execute(GetEventTypeCollectionByNicknameRequest $request): GetEventTypeCollectionByNicknameResponse
+    {
+        $user = $this->userRepository->findByCriteria(
+            new NicknameCriterion($request->getNickname())
+        )->all();
+        $criteria = [new OwnerCriterion($user[0]['id']), new ActiveCriterion()];
```
Comment (aLLeXUs) [inline api/app/Actions/EventType/GetEventTypeCollectionByNicknameAction.php:31]: Add another method for getting a single record (`findOneByCriteria`).

File: api/app/Repositories/User/UserRepository.php [hunk 2]
```diff
@@ -40,4 +42,15 @@ public function markUserEmail(User $user): void
     {
         $user->markEmailAsVerified();
     }
+
+    public function findByCriteria(EloquentCriterion ...$criteria): Collection
```
Comment (aLLeXUs) [inline api/app/Repositories/User/UserRepository.php:46]: The method must be added to the interface too.

File: api/routes/api.php [hunk 3]
```diff
@@ -54,9 +54,16 @@
...
+        Route::get('/nickname/{nickname}', 'EventTypeController@getEventTypesByNickname');
         Route::get('/{id}', 'EventTypeController@getEventTypeById');
     });
 });
+Route::group([
+    'prefix' => '/event-types',
+    'namespace' => 'Api\\'
+], function () {
+    Route::get('/nickname/{nickname}', 'EventTypeController@getEventTypesByNickname');
+});
```
Comment (aLLeXUs) [inline api/routes/api.php:66]: Seems like the wrong group prefix.

**Review by aLLeXUs — CHANGES_REQUESTED** [review]

File: api/routes/api.php (hunk 3 above)
Comment (dmitriy-uvin) [reply api/routes/api.php:66]: But if i move it to group with that same prefix and middleware, i would be unable to fetch event types.
So, as that backend for PUBLIC page, so there mustn't be any middleware.

Comment (aLLeXUs) [reply api/routes/api.php:66]: I do not mean moving to the existing group but moving to a separate route for the public page. But ok, it is up to you.

File: api/app/Repositories/User/UserRepository.php [hunk 4]
```diff
@@ -53,4 +54,15 @@ public function findByCriteria(EloquentCriterion ...$criteria): Collection
 
         return $query->get();
     }
+
+    public function findOneByCriteria(EloquentCriterion ...$criteria): Model
```
Comment (aLLeXUs) [inline api/app/Repositories/User/UserRepository.php:58]: `User` must be returned as it is UserRepository.

**Review by aLLeXUs — CHANGES_REQUESTED** [review]

File: api/app/Repositories/User/UserRepositoryInterface.php [hunk 5]
```diff
@@ -4,12 +4,17 @@
...
+use Illuminate\Support\Collection;
 
 interface UserRepositoryInterface
 {
     public function getById(int $id): ?User;
     public function getByEmail(string $email): ?User;
     public function save(User $user): User;
     public function deleteById(int $id): void;
+    public function findByCriteria(EloquentCriterion ...$criteria): Collection;
+    public function findOneByCriteria(EloquentCriterion ...$criteria): Model;
```
Comment (aLLeXUs) [inline api/app/Repositories/User/UserRepositoryInterface.php:19]: The same for the interface.

File: client/src/store/modules/eventTypes/actions.js [hunk 6]
```diff
@@ -77,5 +77,21 @@ export default {
...
+            );
+            commit(mutations.SET_EVENT_TYPES_BY_NICKNAME, eventTypes);
+            commit('loader/' + loaderMutations.SET_LOADING, false, {
+                root: true
+            });
+        } catch (error) {
+            commit('loader/' + loaderMutations.SET_LOADING, false, {
+                root: true
+            });
+        }
```
[the remark this answers is not in the document]
Comment (dmitriy-uvin) [reply client/src/store/modules/eventTypes/actions.js]: @yevhen-turovets fixed

**Review by aLLeXUs — APPROVED** [review]
Backend looks good

**Review by yevhen-turovets — APPROVED** [review]

**Pull request merged**